server/src/main/java/timely/Server.java

```diff
@@ -142,25 +139,15 @@ private static boolean useEpoll() {
         }
     }
 
-    private static String usage() {
-
-        return "Server <configFile>";
-    }
-
     public static void fatal(String msg, Throwable t) {
         LOG.error(msg, t);
         LATCH.countDown();
     }
 
     public static void main(String[] args) throws Exception {
 
-        if (args.length != 1) {
-            System.err.println(usage());
-        }
-        final File conf = new File(args[0]);
-        if (!conf.canRead()) {
-            throw new RuntimeException("Configuration file does not exist or cannot be read");
-        }
+        Configuration conf = initializeConfiguration(args);
+
         Server s = new Server(conf);
```

Comment:
The server class hasn't changed (but maybe that's what you're complaining about) -- just grabbing the config from Spring in the main method. There is definitely a more Spring-y way to initialize. My goal here was to minimize the disruption, but maybe there's a happy medium with just creating the server class. I'll defer to you guys, but my suggestion would be not to change this now, and then maybe look at another pass to just refactor creation of the Server objects.